src/util/performancetimer.h

```diff
@@ -1,75 +1,39 @@
-/****************************************************************************
-**
-** Copyright (C) 2012 Nokia Corporation and/or its subsidiary(-ies).
-** All rights reserved.
-** Contact: Nokia Corporation (qt-info@nokia.com)
-**
-** This file is part of the QtDeclarative module of the Qt Toolkit.
-**
-** $QT_BEGIN_LICENSE:LGPL$
-** GNU Lesser General Public License Usage
-** This file may be used under the terms of the GNU Lesser General Public
-** License version 2.1 as published by the Free Software Foundation and
-** appearing in the file LICENSE.LGPL included in the packaging of this
-** file. Please review the following information to ensure the GNU Lesser
-** General Public License version 2.1 requirements will be met:
-** http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
-**
-** In addition, as a special exception, Nokia gives you certain additional
-** rights. These rights are described in the Nokia Qt LGPL Exception
-** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
-**
-** GNU General Public License Usage
-** Alternatively, this file may be used under the terms of the GNU General
-** Public License version 3.0 as published by the Free Software Foundation
-** and appearing in the file LICENSE.GPL included in the packaging of this
-** file. Please review the following information to ensure the GNU General
-** Public License version 3.0 requirements will be met:
-** http://www.gnu.org/copyleft/gpl.html.
-**
-** Other Usage
-** Alternatively, this file may be used in accordance with the terms and
-** conditions contained in a signed written agreement between you and Nokia.
-**
-**
-**
-**
-**
-** $QT_END_LICENSE$
-**
-****************************************************************************/
-
 #pragma once
+#include <chrono>
 
-//
-// This is a fork of QPerformanceTimer just without the Q prefix
-// To fix interface changes issues in different QT versions
-// Added restart() function.
-// Returns time in nanosecond resolution.
-//
+#include "util/duration.h"
 
-#include <QtCore/qglobal.h>
+class PerformanceTimer {
+  public:
+    // TODO: make this configurable via a template parameter?
```

Comment:
Don't know if that is a good idea. At least not a todo.

Reply:
Well, it would be trivial and if we ever need to swap out the clock implementation using ifdefs, it'd prefer if it was done by `#ifdef`ing `using PerformanceTimer = PerformanceTimerGeneric</* clock impl */>` instead of in the class body. That's purely a style choice however. So lets KISS and just remove the comment?